packages/story-editor/src/components/colorPicker/colorPicker.js

```diff
@@ -113,15 +114,15 @@ function ColorPicker({
     [onDebouncedChange, selectedElementIds, changedStyle, pushTransform]
   );
 
-  const closeIfNotEyedropping = () => {
-    if (!isEyedropperActive) {
+  const maybeClose = () => {
+    if (!isEyedropperActive && !showDialog) {
       onClose();
     }
   };
```

Comment:
I think this deserves a comment or two - it looks pretty simple, but the reasoning can be a bit deceptive.